python/lsst/ts/scriptqueue/ui/request_model.py

```diff
@@ -0,0 +1,547 @@
+
+        Returns
+        -------
+        available_scripts : dict
+            A dictionary with "external" and "standard" list of scripts.
+
+        """
+
+        script_list = self.queue.evt_availableScripts.get()
+        if script_list is None:
+            self.log.debug("No previous information about script list. Requesting.")
+            available_scripts_coro = self.queue.evt_availableScripts.next(flush=False,
+                                                                          timeout=self.cmd_timeout)
+
+            self.run(self.queue.cmd_showAvailableScripts.start(timeout=self.cmd_timeout))
+
+            script_list = self.run(available_scripts_coro)
+
+        return {'external': script_list.external.split(':'),
+                'standard': script_list.standard.split(':')}
+
+    def pause_queue(self):
+        """Pause the queue."""
+        self.run(self.queue.cmd_pause.start())
+
+    def resume_queue(self):
+        """Resume queue operation"""
+        self.run(self.queue.cmd_resume.start())
+
+    def quit_queue(self):
+        """Quit queue by sending it to exit control. Assume the queue is in enable state
+        but if it is not, will skip the rejected commands for disable and standby.
+        """
+
+        try:
+            self.run(self.queue.cmd_disable.start(timeout=self.cmd_timeout))
+        except Exception as e:
+            pass
+
+        try:
+            self.run(self.queue.cmd_standby.start(timeout=self.cmd_timeout))
+        except Exception as e:
+            pass
+
+        try:
+            self.run(self.queue.cmd_exitControl.start(timeout=self.cmd_timeout))
+        except Exception as e:
+            pass
+
+        return True
+
+    def stop_scripts(self, sal_indices, terminate=False):
+        """Stop list of scripts.
+
+        Parameters
+        ----------
+        sal_indices : list(int)
+            A list of script indices to stop.
+
+        terminate : bool
+            Should the script be terminated?
+
+        """
+        self.queue.cmd_stopScripts.set(salIndices=sal_indices,
+                                       length=len(sal_indices),
+                                       terminate=terminate)
+        self.run(self.queue.cmd_stopScripts.start(timeout=self.cmd_timeout))
+
+    def get_queue_state(self):
+        """A method to get the state of the queue and the scripts running on the queue.
+
+        Returns
+        -------
+        state : dict
+
+        """
+        self.update_queue_state()
+
+        state = {'state': self.state['state'],
+                 'length': self.state['length'],
+                 'past_length': self.state['past_length'],
+                 'queue_scripts': {},
+                 'past_scripts': {},
+                 'current': None}
+
+        for i in range(self.state['length']):
+            state['queue_scripts'][self.state['queue_scripts'][i]] = \
+                self.get_script_info(self.state['queue_scripts'][i])
```

Comment:
I strongly suggest you make state an object, not a dict, e.g.:
```
class QueueState:
   def __init__(self):
         ....
    def update(self, queue):
       """Update using the current value of the `ScriptQueue` `queue` event."""
       ...
```
That way you can document what the fields are. I have no idea what "queue_scripts" is anymore. I though it was just a list of script SAL indices, but based on the next loops maybe not?